dvc/repo/metrics/show.py

```diff
@@ -130,14 +130,17 @@ def _read_metrics(self, metrics, branch):
     for out, typ, xpath in metrics:
         assert out.scheme == "local"
         if out.use_cache:
+            path = self.cache.local.get(out.checksum)
```

Comment:
This is not going to work, as it will return a path that looks like `.dvc/cache/12/3456789`, where `123456789` is a md5 of that file. We should use `out.path` in _check_fiile_type instead.

Reply:
Thanks! I have fixed it.